pygnmi/client.py

```diff
@@ -673,20 +680,35 @@ def set(
                 )
 
             if gnmi_extension:
-                gnmi_message_request = SetRequest(
-                    prefix=protobuf_prefix,
-                    delete=del_protobuf_paths,
-                    update=update_msg,
-                    replace=replace_msg,
-                    extension=[gnmi_extension],
-                )
+                if prefix is None and target is None:
+                    gnmi_message_request = SetRequest(
+                        delete=del_protobuf_paths,
+                        update=update_msg,
+                        replace=replace_msg,
+                        extension=[gnmi_extension],
+                    )
+                else:
+                    gnmi_message_request = SetRequest(
+                        prefix=protobuf_prefix,
+                        delete=del_protobuf_paths,
+                        update=update_msg,
+                        replace=replace_msg,
+                        extension=[gnmi_extension],
+                    )
```

Comment:
I am assuming you mean ```not (prefix and target)```. That is logically incorrect. The correct condition would be ```(a or b)```. But this wouldn't work because we need to specifically check if the value is not None, since an empty string evaluates to False as well.